sorting.go

```diff
@@ -13,48 +13,55 @@ type peerDistance struct {
 	distance ID
 }
 
-// peerSorterArr implements sort.Interface to sort peers by xor distance
-type peerSorterArr []*peerDistance
+// peerDistanceSorter implements sort.Interface to sort peers by xor distance
+type peerDistanceSorter struct {
+	peers  []peerDistance
+	target ID
+}
 
-func (p peerSorterArr) Len() int      { return len(p) }
-func (p peerSorterArr) Swap(a, b int) { p[a], p[b] = p[b], p[a] }
-func (p peerSorterArr) Less(a, b int) bool {
-	return p[a].distance.less(p[b].distance)
+func (pds peerDistanceSorter) Len() int      { return len(pds.peers) }
+func (pds peerDistanceSorter) Swap(a, b int) { pds.peers[a], pds.peers[b] = pds.peers[b], pds.peers[a] }
+func (pds peerDistanceSorter) Less(a, b int) bool {
+	return pds.peers[a].distance.less(pds.peers[b].distance)
 }
 
-//
+// Append the peer.ID to the sorter's slice. It may no longer be sorted.
+func (pds peerDistanceSorter) appendPeer(p peer.ID) peerDistanceSorter {
```

Comment:
It would improve the ergonomics and probably won't cost us anything to pass the receiver by pointer and modify this in place. `sort.Sort` will allocate this anyways (unless escape analysis does something fancy) when converting this to an interface.

Reply:
Sounds right. You mean to remove the return value as well? Having a pointer receiver in peerDistanceSorter.sort may also remove an allocation (trying to jam peerDistanceSorter by value into an interface, as opposed to allowing a pointer).

Reply:
Yep.